app/javascript/shared/components/tags.jsx

```diff
@@ -334,11 +334,7 @@ export class Tags extends Component {
       }
       // updates searchResults array according to what is being typed by user
       // allows user to choose a tag when they've typed the partial or whole word
-      this.setState({
-        searchResults: response.result.filter(
-          (t) => !this.selected.includes(t.name),
-        ),
```

Comment:
The only issue with this is that my previous selections still continue to pop in the search results (as well as the one I'm currently typing/searching)

Maybe instead of 

```
searchResults: response.result.filter(
          (t) => !this.selected.includes(t.name),
        ),
```

We could filter out any selected that aren't the current query:

```
searchResults: response.result.filter(
          (t) => t.name === query || !this.selected.includes(t.name),
        ),
```

Then when I have `java` already selected and start my next search with `java`, I only see `javascript`

Reply:
Good catch! Tbh this was the quickest thing that worked to address the original problem I didn't check too deeply for other potential issues yet, which is why I had just opened a draft before heading out on Thursday.

Reply:
Unfortunately, this doesn't work by itself either. For example, when I add "java" as a tag, it will no longer be in the autocomplete options when looking for "javascript" but will pop up again once I get to "java" because now the "t.name === query` part is true.

Reply:
This seems to fix it:

```js
(t) => (t.name === query && !this.selected.includes(t.name)) || !this.selected.includes(t.name),
```